indico/modules/users/tasks.py

```diff
@@ -37,3 +60,211 @@ def update_gravatars(user=None):
             continue
         set_user_avatar(user, gravatar, 'gravatar', lastmod)
         logger.info('Gravatar of user %s updated', user)
+
+
+def export_user_data(user, options):
+    """Generate a zip file with all user data and save it in user.data_export_request."""
+    export_request = user.data_export_request
+    if export_request and export_request.is_running:
+        return
+
+    # ughh why isn't there an ORM upsert..
+    if export_request:
+        db.session.delete(export_request)
+        db.session.commit()
+    export_request = DataExportRequest(user=user, selected_options=options, state=DataExportRequestState.running)
+    db.session.commit()
+
+    temp_file = NamedTemporaryFile(suffix='.zip', dir=config.TEMP_DIR, delete=False)
+    try:
+        _export_user_data(export_request, temp_file)
+    except Exception:
+        logger.exception('Could not create a user data export %r', export_request)
+        export_request.fail()
+        db.session.commit()
+        notify_data_export_failure(export_request)
+    finally:
+        Path(temp_file.name).unlink(missing_ok=True)
+
+
+def _export_user_data(export_request, buffer):
+    generate_zip(export_request, buffer, config.MAX_DATA_EXPORT_SIZE * 1024**2)
+    buffer.seek(0)
+
+    file = File(filename='data-export.zip', content_type='application/zip')
+    try:
+        file.save(('user', export_request.user.id), buffer)
+        file.claim()
```

Comment:
We delete unclaimed files after 1 day, so if the file is claimed we need some custom logic to delete those archives after a certain time (1 week maybe or when requesting a new one?)